### PR TITLE
Add Cargo team charter.

### PR DESCRIPTION
**Pull request by ehuss**
This adds a document explaining what the Cargo team is and how it operates.

This is based on a document I prepared a couple years ago during the efforts to have charters for all teams. That effort has stalled, but I wanted to capture these ideas and make it clearer what is involved with being a member and how to join. The only section that is significantly new is the "Decision process" section.

In terms of detail, I wanted to strike a balance between providing the important information (the "scaffolding" so to speak), versus an exhaustive legal document. For example, this doesn't have specific requirements on what "involvement" means, team limits, term limits, specific [vacation policy](https://github.com/rust-lang/governance/blob/master/common/vacation_policy.md), accountability, consensus and conflict resolution, non-requirements (things not expected of members), specific deliverables or outcomes, measurements of "success", complete listing of the code and systems the team is responsible for, cross-team relationships, sub-team and working-group creation, [roadmap management](https://rust-lang.github.io/rfcs/3037-roadmap-2021.html), ...etc.

For similar information, see:
* https://rustc-dev-guide.rust-lang.org/compiler-team.html
* https://std-dev-guide.rust-lang.org/team.html
* https://rust-lang.github.io/rfcs/1068-rust-governance.html
* https://rust-lang.github.io/rfcs/2689-compiler-team-contributors.html
* [Libs impl sample charter](https://hackmd.io/@rust-libs/HkgFCFWku)

### CODE REVIEW

File: src/doc/contrib/src/team.md
Comment (weihanglo) [inline src/doc/contrib/src/team.md:98]: I wonder which category adding a new dependency falls into. We've discussed this from time to time. Is it worth a mention?

**Review by weihanglo — APPROVED** [review]
Very complete and informative. It is much appreciated!

File: src/doc/contrib/src/team.md
Comment (epage) [inline src/doc/contrib/src/team.md:125]: We have added some commands with a change proposal, rather than an FCP.  Should we make a distinction on this?

I'm assuming RFC is more needed when the command has non-trivial workflows and hasn't been broadly adopted by the community to vet those workflows. 
- `cargo rm` is more trivial
- `cargo add` is a little less trivial but had broad use for a long time
- `cargo-script` has non-trivial workflows and isn't pervasive.  Even if it was, the workflows are non-trivial enough to warrant an RFC to re-evaluate those design decisions
  - I think `cargo-nextest` is a good example of this also.  It has done a lot to improve workflows but in a way that works around limitations in the ecosystem.  Instead of replacing `cargo test` with it, we'd need to decide what non-workaround solutions to the problems look like

**Review by epage — COMMENTED** [review]
Thanks!

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: Are there any specific details that you are thinking of? The only specific thing I can imagine is maybe notifications of security issues, but that's kind of the purview of the security team, and is already somewhat covered by being "a contact point with other teams".  

If there are specific things you'd like to see here, let me know.

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: I intentionally wasn't wanting to make it a link, just to avoid the temptation to actually click it, but that is a bad motivation.

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md:98]: That's a good question. I would kind of like to make that a separate thing, outside of this chapter and PR.  I'm not sure where it would go, but I think it would be great to have written guidelines for dependencies. I'm wondering if there are other guidelines that are worth writing down? Like maybe review guidelines? Test requirements? Guidelines for when to revert?

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: Sounds good, I merged the sections.

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: Add a link.

File: src/doc/contrib/src/team.md
Comment (weihanglo) [reply src/doc/contrib/src/team.md]: It's more like “if you were about to step down today, what would you want the new team lead to know.” 

I don't have any specific idea on that, just feel like knowing more details can make the team more robust. If there is nothing new that's fine!

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: I split the list up and tried to make it clearer up-front that these aren't all requirements.  I wasn't quite sure how to structure it, so let me know if you have any ideas to clean it up.

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md:47]: Can you say more about what your thoughts are here? I think I have a vague idea of what you're getting at here, but I'm not sure how to put it into words. I assume this is trying to make the distinction of making some public statement, and wanting to distinguish between "I'm speaking for the whole team" and "I'm just giving my personal viewpoint"?

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: Yea, this is a bit of a gray area.  I added some extra text here, though I'm not sure if it is really complete. Let me know if you have any suggestions to the wording.

File: src/doc/contrib/src/team.md
Comment (ehuss) [reply src/doc/contrib/src/team.md]: I can't think of too many specific things.
I assume the next leader would also be added as a member to the devtools team. Right now that doesn't really involve much other than monitoring the devtools zulip channel.

File: src/doc/contrib/src/team.md
Comment (weihanglo) [inline src/doc/contrib/src/team.md:49]: One thing missing here is the permission of crates.io for crates maintained by the Cargo team. Who has the right to publish manually? Or should we make all things in automation even for a point release of cargo-the-library?

File: src/doc/contrib/src/team.md
Comment (epage) [reply src/doc/contrib/src/team.md:47]: Thats it. I'm not sure what to even put down for a policy, if anything. Its just something I try to keep in mind

File: src/doc/contrib/src/team.md
Comment (epage) [reply src/doc/contrib/src/team.md:49]: If we adopt cargo-release, it would semi-automate the process. One of the more recent features is you can set a workspace-wide cargo-owner and it will be set when publishing new crates. I added this when I I realized there were newer clap crates without the team added to them.

Comment (ehuss) [reply src/doc/contrib/src/team.md:49]: At this time I don't believe the infra team is willing to support tokens for auto-publishing from CI.  They are at least waiting on scoped tokens. 

I'm not sure how best to manage crates.io ownership. I think for now people can be added on an as-needed basis. I think we should preferably keep the ownership list small, perhaps eventually with no individual owners.  I'm not sure if we can fully automate it once scoped tokens are available (such as figuring out how to trigger releases, etc.).

File: src/doc/contrib/src/team.md
Comment (weihanglo) [reply src/doc/contrib/src/team.md:47]: We would always be recognized as “Cargo team member” or whatever your strongest public image when interacting with the community. That's a thing one can never shave off even after stepping down. 

For example, when one team member talks about `cargo-eval` pre-RFC, even when they state it's their own personal opinion. They still have a vote to affect the decision. People then tend to see it as a portion of the team's opinion if they find the connection.

That's an inevitable consequence of being a formal member.

Comment (weihanglo) [reply src/doc/contrib/src/team.md:47]: In terms of making a statement for the whole team, we usually start like “after the team's recent discussion, we….” I think that kind of speaking gives enough information to the audience to decide where it comes from. 

Just be clear. I am fine with not having a policy here for now.

**Review by arlosi — APPROVED** [review]

**Comment by ehuss** [comment]
I want to make sure there is a unanimous approval for adding this, so I'd like to make sure that this isn't merged until everyone checks off on it.

@rfcbot fcp merge

File: src/doc/contrib/src/team.md
Comment (joshtriplett) [inline src/doc/contrib/src/team.md:19]:
```suggestion
The leader is selected by consensus of the existing members with no objections.
```

File: src/doc/contrib/src/team.md
Comment (joshtriplett) [reply src/doc/contrib/src/team.md:47]: It's absolutely the case that we'll be perceived as Cargo team members, but with some care we can make sure to state when we're speaking for the team, and when we're specifically *not* speaking for the team.

File: src/doc/contrib/src/team.md
Comment (joshtriplett) [reply src/doc/contrib/src/team.md:49]: > At this time I don't believe the infra team is willing to support tokens for auto-publishing from CI. They are at least waiting on scoped tokens.

Experimental support for scoped tokens has shipped, so it's possible to do this now.

File: src/doc/contrib/src/team.md
Comment (joshtriplett) [inline src/doc/contrib/src/team.md:67]:
```suggestion
They can be added by unanimous consent of the team.
The team lead or another member of the team will also confirm with the moderation team that there are no concerns involving the proposed team member.
```

**Comment by joshtriplett** [comment]
@rfcbot concern consensus-with-no-objections
@rfcbot concern check-with-moderation-team-on-proposed-members

@rfcbot reviewed

**Comment by ehuss** [comment]
@joshtriplett Thank you very much for the review, I have applied your suggestions. (I don't think fcpbot allows me to resolve your concerns, but I don't think it is strictly necessary since they are just applying your suggestions.)

I'm going to go ahead and merge. Thank you everyone for the review. If you have any further questions or concerns, feel free to comment, open a new PR or issue, ask on Zulip, or direct message me. 

The intent going forward is that any significant changes to this document should go through an FCP (minor clarifications, or editorial changes don't require that). This first version was intended to be a starting baseline where we can then update when we identify things that would be helpful.

@bors r+

File: src/doc/contrib/src/team.md
Comment (Eh2406) [reply src/doc/contrib/src/team.md:47]: Following today's episode of the rust drama... being unclear about whether someone was speaking on behalf of the team or just expressing their own opinion may have been a key contributor to this round of drama.

**Pull request merged**
